src/compiler/scala/tools/nsc/transform/UnCurry.scala

```diff
@@ -213,19 +213,23 @@ abstract class UnCurry extends InfoTransform
         // Expand the function body into an anonymous class
         gen.expandFunction(localTyper)(fun, inConstructorFlag)
       } else {
+        val mustExpand = mustExpandFunction(fun)
+        val newNamePrefix = if (mustExpand) nme.DELAMBDAFY_LAMBDA_CLASS_NAME else nme.ANON_FUN_NAME
```

Comment:
I need to document this. I needed to exclude non LMF functions/SAMs from the static-rewrite in `Delambdafy`, which runs for methods that satisfy `isDelambdafyTarget`.

Ideally we'd be able to rewrite them as well, but it gets more involved as we need to rewrite the anonymous class that delegates to this method in a non-trivial way.

Maybe a tree or symbol attachment would be a cleaner way to implement `isDelambdafyTarget`.